baseapp/queryrouter.go

```diff
@@ -21,16 +22,22 @@ func NewQueryRouter() *QueryRouter {
 
 // AddRoute adds a query path to the router with a given Querier. It will panic
 // if a duplicate route is given. The route must be alphanumeric.
-func (qrt *QueryRouter) AddRoute(path string, q sdk.Querier) sdk.QueryRouter {
-	if !sdk.IsAlphaNumeric(path) {
+func (qrt *QueryRouter) AddRoute(route string, q sdk.Querier) sdk.QueryRouter {
+	if !sdk.IsAlphaNumeric(route) {
 		panic("route expressions can only contain alphanumeric characters")
 	}
 
-	if qrt.routes[path] != nil {
-		panic(fmt.Sprintf("route %s has already been initialized", path))
+	// paths are only the final extensions!
+	// Needed to ensure erroneous queries don't get into the state machine.
+	if strings.Contains(route, "/") {
+		panic("route's don't contain '/'")
```

Comment:
Since I imported the code from cosmos-sdk, I don't think there is any reason to change the comments arbitrarily. If it is changed to a better annotation later, how about editing it then?